src/test/java/uk/nhs/hee/tis/trainee/sync/service/PlacementSyncServiceTest.java

```diff
@@ -114,7 +130,14 @@ void shouldNotFindRecordByIdPostWhenNotExists() {
   }
 
   @Test
-  void shouldSendRetrievalRequest() {
-    assertThrows(UnsupportedOperationException.class, () -> service.request(ID));
+  void shouldEnrichARequestPost() throws JsonProcessingException {
+    when(postSyncService.findById(ID)).thenReturn(Optional.<Post>empty());
+
+    Map<String, String> recordData = record.getData();
+    recordData.put("postId", ID);
+
+    service.enrichOrRequestPost(record);
+
+    verify(postSyncService).request(ID);
```

Comment:
Need to implement and test the `request` method, other changes should be reverted.